lib/execution/executor.ex

```diff
@@ -55,15 +55,18 @@ defmodule WaspVM.Executor do
   # What happens is we pass in the main limit for the gas & the gas_limit,
   # then every iteration before we procedd we check the gas limit and the
   # returned op_gas (gas accumulted from executing that opcode)
-  def execute(frame, vm, gas, stack, total_instr, gas_limit, next_instr \\ 0)
-  def execute(_frame, vm, gas, stack, _total, gas_limit, _next) when gas_limit != :infinity and gas > gas_limit, do: IEx.pry #{:error, :reached_gas_limit}
-  def execute(_frame, vm, gas, stack, total_instr, _gas_limit, next_instr) when next_instr >= total_instr or next_instr < 0, do: {vm, gas, stack}
-  def execute(frame, vm, gas, stack, total_instr, gas_limit, next_instr) do
+  def execute(frame, vm, gas, stack, total_instr, gas_limit, trace, next_instr \\ 0)
+  def execute(_frame, vm, gas, stack, _total, gas_limit, trace, _next) when gas_limit != :infinity and gas > gas_limit, do: IEx.pry #{:error, :reached_gas_limit}
+  def execute(_frame, vm, gas, stack, total_instr, _gas_limit, trace, next_instr) when next_instr >= total_instr or next_instr < 0, do: {vm, gas, stack}
+  def execute(frame, vm, gas, stack, total_instr, gas_limit, trace, next_instr) do
     %{^next_instr => instr} = frame.instructions
-
     {{frame, vm, next_instr}, gas, stack} = instruction(instr, frame, vm, gas, stack, next_instr)
 
-    execute(frame, vm, gas, stack, total_instr, gas_limit, next_instr + 1)
+    if trace == true do
```

Comment:
We should never be comparing to a boolean in any expression, since the expression already gets evaluated to a boolean. This should be rewritten as
```elixir
if trace, do: write_to_file(instr, gas)
```

Reply:
Fixed